src/hooks/useOutputMaterial.ts

```diff
@@ -63,24 +58,39 @@ export function useOutputMaterial(sessionId: string, outputIdOrName: string, mat
 
 		// get all geometry and materials below the node
 		const geometryData = getGeometryData(node);
-		const materials = getMaterials(geometryData);
-	
-		if (materials.length > 0) {
-			// update existing materials
-			materials.forEach((m) => {
-				for ( const p in materialProps)
-					(<any>m)[p as keyof MaterialStandardData] =
-					materialProps[p as keyof IMaterialStandardDataProperties];
-				m.updateVersion();
-			});
-		}
-		else
-		{
-			// no material found: define new material
-			const material = new MaterialStandardData(materialProps);
-			geometryData.forEach(data => data.material = material);
+
+		let MaterialClassType: typeof MaterialStandardData | typeof MaterialSpecularGlossinessData | typeof MaterialUnlitData | typeof MaterialGemData;
+		switch (materialType) {
+		case MaterialType.SpecularGlossiness:
+			MaterialClassType = MaterialSpecularGlossinessData;
+			break;
+		case MaterialType.Unlit:
+			MaterialClassType = MaterialUnlitData;
+			break;
+		case MaterialType.Gem:
+			MaterialClassType = MaterialGemData;
+			break;
+		default:
+			MaterialClassType = MaterialStandardData;
 		}
 
+		// update all geometry materials
+		geometryData.forEach(data => {
+			if (data.material && data.material instanceof MaterialClassType) {
+				// update existing material
+				for ( const p in materialProps)
+					(<any>data.material)[p as keyof typeof MaterialClassType] 
+					= materialProps[p as keyof typeof materialProps];
+
+				data.material.updateVersion();
+			} else {
+				// no material found: define new material
+				const material = new MaterialClassType(materialProps);
+				data.material = material;
```

Comment:
@MajorMeerkatThe3rd ony thing I did not understand yet: Is it necessary to create a separate material instance for each element of `geometryData`? Could we create one instance and use it for all of them?

Reply:
@snabela I just overlooked that. done in commit https://github.com/shapediver/ShapeDiverReactExample/pull/47/commits/6f952861af36ee4c8f31b469507907176c4f1c29